crates/sncast/tests/helpers/fixtures.rs

```diff
@@ -266,35 +270,116 @@ pub fn create_test_provider() -> JsonRpcClient<HttpTransport> {
-        .expect("Failed to create Utf8PathBuf from PathBuf");
 
     fs_extra::dir::copy(
-        src_dir.join("src"),
-        &dest_dir,
-        &fs_extra::dir::CopyOptions::new().overwrite(true),
+        src_dir.as_ref(),
+        temp_dir.as_ref(),
+        &fs_extra::dir::CopyOptions::new().content_only(true),
     )
-    .expect("Unable to copy the src directory");
-    fs_extra::file::copy(
-        src_dir.join("Scarb.toml"),
-        dest_dir.join("Scarb.toml"),
-        &fs_extra::file::CopyOptions::new().overwrite(true),
-    )
-    .expect("Unable to copy Scarb.toml");
+    .expect("Failed to copy directory content");
+
+    temp_dir
+}
+
+#[must_use]
+pub fn duplicate_contract_directory_with_salt(
+    src_dir: impl AsRef<Utf8Path>,
+    code_to_be_salted: &str,
+    salt: &str,
+) -> TempDir {
+    let src_dir = Utf8PathBuf::from(src_dir.as_ref());
+
+    let temp_dir = copy_dir_to_temp_dir(&src_dir);
 
+    let dest_dir = Utf8PathBuf::from_path_buf(temp_dir.path().to_path_buf())
+        .expect("Failed to create Utf8PathBuf from PathBuf");
+
+    let file_to_be_salted = "src/lib.cairo";
     let contract_code =
-        fs::read_to_string(src_dir.join("src/lib.cairo")).expect("Unable to get contract code");
-    let updated_code = contract_code.replace(to_be_salted, &(to_be_salted.to_string() + salt));
-    fs::write(dest_dir.join("src/lib.cairo"), updated_code)
+        fs::read_to_string(src_dir.join(file_to_be_salted)).expect("Unable to get contract code");
+    let updated_code =
+        contract_code.replace(code_to_be_salted, &(code_to_be_salted.to_string() + salt));
+    fs::write(dest_dir.join(file_to_be_salted), updated_code)
         .expect("Unable to change contract code");
 
     temp_dir
 }
 
+#[must_use]
+pub fn duplicate_script_directory(
+    src_dir: impl AsRef<Utf8Path>,
+    deps: Vec<impl AsRef<std::path::Path>>,
+) -> TempDir {
+    let mut deps = get_deps_map_from_paths(deps);
+
+    let src_dir = Utf8PathBuf::from(src_dir.as_ref());
+
+    let temp_dir = copy_dir_to_temp_dir(&src_dir);
+
+    let dest_dir = Utf8PathBuf::from_path_buf(temp_dir.path().to_path_buf())
+        .expect("Failed to create Utf8PathBuf from PathBuf");
+
+    let manifest_path = dest_dir.join("Scarb.toml");
+    let contents = fs::read_to_string(&manifest_path).unwrap();
+    let mut parsed_toml: Table = toml::from_str(&contents)
+        .with_context(|| format!("Failed to parse {manifest_path}"))
+        .unwrap();
+
+    let deps_toml = parsed_toml
+        .get_mut("dependencies")
+        .unwrap()
+        .as_table_mut()
+        .unwrap();
+
+    if let Some(sncast_std) = deps_toml.get_mut("sncast_std") {
+        if let Some(sncast_std_path) = sncast_std.get_mut("path") {
+            let sncast_std_path =
+                Utf8PathBuf::from(sncast_std_path.as_str().expect("Failed to extract string"));
+            if sncast_std_path.is_relative() {
```

Comment:
this will always be relative in our case, + you probably can just canonicalize absolute paths as well, which might simplify the function

Reply:
https://github.com/foundry-rs/starknet-foundry/pull/1622/files#diff-382c29dde6aa65ca0c28f84a85375ae89c899870bd66f1a520f5ade0e6d25e9dR341

src_dir may potentially be absolute, I don't think joining two absolute paths will return existing path

Reply:
wait, you have linked a path inside this if, not sure what you meant by that
I think in this case src_dir will always be relative, since we only test with latest (not-released) sncast_std, right? we do not test scripts that have sncast_std set to urls

Reply:
we don't, but now we can if we want :)

Reply:
but, there's not point, right? :P

Reply:
refactored